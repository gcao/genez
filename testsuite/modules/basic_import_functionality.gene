#!/usr/bin/env gene run

# Test actual module import functionality
# Create an in-memory module using namespace
(ns math
  (fn square [x] (* x x))
  (fn cube [x] (* x x x)))

# Access module members
(print "Square of 5: " (math/square 5))
(print "Cube of 3: " (math/cube 3))