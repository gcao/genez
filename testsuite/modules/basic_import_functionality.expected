Square of 5:  25
Cube of 3:  27